examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1520 @@
+    "    transform = transforms.Compose([\n",
+    "        transforms.Resize(256),\n",
+    "        transforms.CenterCrop(224),\n",
+    "        transforms.ToTensor(),\n",
+    "        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We load the resnet50 model remove the last layer, so the output is then the second last layer with dimension 1x2048.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['all_X.lib']"
+      ]
+     },
+     "execution_count": 28,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model = models.resnet50(pretrained=True)\n",
+    "modules = list(model.children())[:-1]\n",
+    "model=torch.nn.Sequential(*modules)\n",
+    "for p in model.parameters():\n",
+    "    p.requires_grad = False\n",
+    "    \n",
+    "model.eval()\n",
+    "#apply modified resnet50 to data\n",
+    "dataloaders = DataLoader(mimicDataset(input_list[:n,:]), batch_size=n,num_workers=0)\n",
+    "    \n",
+    "data, labels = next(iter(dataloaders))\n",
+    "with torch.no_grad():\n",
+    "    features_var = model(data)\n",
+    "    features = features_var.data \n",
+    "    all_X = features.reshape(n,2048).numpy()\n",
+    "    \n",
+    "#save feature matrix\n",
+    "dump(all_X, \"all_X.lib\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finetuning a pretrained CNN and extracting the second last layer as features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the second approach of image encoding, we use the concept of transfer learning. \n",
+    "Therefore, we take a pretrained CNN and continue training the model with our data. This saves a lot of time and leads to satisfying results even on a small dataset. \n",
+    "For the implementation we again use pytorch and the pretrained resnet50 from torchvision. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we split the data according to the split given in mimic-cxr-jpg. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_train = input_list[input_list[:,5] == 'train',:]\n",
+    "input_validate = input_list[input_list[:,5] == 'validate',:]\n",
+    "input_test = input_list[input_list[:,5] == 'test',:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the dataset is unbalanced, we use a weighted sampler "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class_counts = np.zeros(14)\n",
+    "for i in range(14): class_counts[i] = sum(input_train[:,6]==i)\n",
```

Comment:
```suggestion
    "for i in range(num_classes):"
    "     class_counts[i] = sum(input_train[:,6]==i)\n",
```

I remember there is no `num_classes`parameter defined, but I strongly recommend you to define it :) usually it's better not to use any hardcode parameters (e.g. `14`), but to introduce them once in the beginning and use the variables instead